### PR TITLE
front: rs-editor: display power class

### LINKED ISSUE #5210: front: display power classes associated with existing RS power restriction codes
## Description

When editing/creating a rollingstock, in the curves tab, each power restriction code (if existing) should have its associated power class displayed next to it.
The user need to be able to modify the power class and save the changes

image

## AC

- [ ] Power class should be an input displayed next to each power restriction code
- [ ] After updating a power class and saving the changes, the new power class should be displayed in the rolling stock curves tab

### LINKED ISSUE #5256: front: rolling stock editor - power restricton codes disappearing
### What happened?

In the rolling stock editor, when a rolling stock has power restriction codes, they are displayed on the right side of the screen.

[Capture d’écran du 2023-10-09 15-14-13.odg](https://github.com/osrd-project/osrd/files/12846530/Capture.d.ecran.du.2023-10-09.15-14-13.odg)

If we try to update one of the characteristic of the rolling stock, the list disappears.

[Capture d’écran du 2023-10-09 15-07-21.odg](https://github.com/osrd-project/osrd/files/12846488/Capture.d.ecran.du.2023-10-09.15-07-21.odg)

### What did you expect to happen?

The power restriction codes list of the rolling stock should remain displayed

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the rolling stock editor
2. Select a duplicated rolling stock (or duplicate one first)
3. Change of the parameters and validate the change

### What operating system, browser and environment are you using?

- **Browser**: Firefox v117
- **OS**: Ubuntu
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

093127c1

### LINKED ISSUE #5281: front: disable the RP code's power class "Non renseignée" and contain the default value
## Description

When editing/creating a rolling stock, in the curves tab, the "Non specified" power restriction (PR) should have the base power class displayed next to it

## AC

- [ ] The power class displayed next to the "Non specified" PR should be the base power class (defined in general characteristics tab)
- [ ] The field should be disabled
- [ ] If the base power class is not specified, the filed in curves tab should be empty

### PR DESCRIPTION
**Pull request by SharglutDev**
display power class next to its associated power restriction code in the curves tab

close #5210 
close #5281 
close #5256

### CODE REVIEW

File: front/src/common/SelectorSNCF.tsx [hunk 1]
```diff
@@ -35,53 +49,82 @@ export default function SelectorSNCF<T extends string | null, K extends string>(
     onItemRemoved,
     translationFile,
     translationList = '',
+    extraColumn,
   } = props;
   const { t } = useTranslation(translationFile);
 
   return (
-    <div className={`${mainClass}`}>
+    <div className={`${mainClass}${extraColumn ? '-withExtraColumn' : ''}`}>
```
Comment (clarani) [inline front/src/common/SelectorSNCF.tsx]: I find this clearer, what do you think ?
```suggestion
    <div className={extraColumn ? `${mainClass}-withExtraColumn` : `${mainClass}`}>
```

File: front/src/common/SelectorSNCF.tsx [hunk 2]
```diff
@@ -35,53 +49,82 @@ export default function SelectorSNCF<T extends string | null, K extends string>(
...
-                  {!isNull(item)
-                    ? t(getTranslationKey(translationList, String(item)))
-                    : t('unspecified')}
-                </div>
-                {((permanentItems && !permanentItems.includes(item)) || !permanentItems) &&
-                  onItemRemoved && (
-                    <div className={`${mainClass}-trash-icon`}>
-                      <FaTrash onClick={() => onItemRemoved(item, title)} />
-                    </div>
+              <div className={`${extraColumn && `d-flex ml-1`}`} key={`selector-${title}-${index}`}>
```
Comment (clarani) [inline front/src/common/SelectorSNCF.tsx]:
```suggestion
              <div className={cx({ 'd-flex ml-1': extraColumn }) key={`selector-${title}-${index}`}>
```

File: front/src/common/SelectorSNCF.tsx [hunk 3]
```diff
@@ -35,53 +49,82 @@ export default function SelectorSNCF<T extends string | null, K extends string>(
...
+                  className={cx(
+                    `${mainClass}-item`,
+                    `${extraColumn && 'withExtraColumn'}`,
+                    'd-flex',
+                    'mb-1',
+                    borderClass,
+                    {
+                      selected: item === selectedItem,
+                      hovered: item === hoveredItem && item !== null,
+                    }
```
Comment (clarani) [inline front/src/common/SelectorSNCF.tsx]:
```suggestion
                 className={cx(`${mainClass}-item`, 'd-flex', 'mb-1', borderClass, {
                    withExtraColumn: extraColumn,
                    selected: item === selectedItem,
                    hovered: item === hoveredItem && item !== null,
                  })}
```

File: front/src/modules/rollingStock/consts.ts [hunk 4]
```diff
@@ -55,7 +60,7 @@ export const effortCurveCondKeys: EffortCurveCondKeys = {
 };
 
 export type RollingStockParametersValues = {
-  [key: string]: string | number | null | EffortCurves;
+  [key: string]: string | number | null | EffortCurves | RollingStockBase['power_restrictions'];
```
Comment (clarani) [inline front/src/modules/rollingStock/consts.ts:64]: Can you add this comment please ? For the refacto :)
```suggestion
  // TODO: remove this line in the type
  [key: string]: string | number | null | EffortCurves | RollingStockBase['power_restrictions'];
```

File: front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorCurves.tsx [hunk 5]
```diff
@@ -429,14 +435,13 @@ export default function RollingStockEditorCurves({
     title: K
   ) => {
     if (!currentRsEffortCurve) return;
+    const rsEffortCurveCondKey = effortCurveCondKeys[title as keyof EffortCurveCondKeys];
```
Comment (clarani) [inline front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorCurves.tsx:438]: If you change the signature of the function like this (replace RollingStockSelectorParam by EffortCurveCondKeys), you don't need the cast anymore
```
 const removeAnotherRsParam = <K extends keyof EffortCurveCondKeys>(
    value: EffortCurveCondKeys[K],
    title: K
  ) => {}
```
and then 
```suggestion
    const rsEffortCurveCondKey = effortCurveCondKeys[title];
```

File: front/public/locales/fr/rollingstock.json [hunk 6]
```diff
@@ -79,7 +79,8 @@
   "notLocked": "Non verrouillé",
   "number": "Numéro",
   "powerClass": "Classe puissance",
-  "powerRestrictions": "Restrictions de puissance",
+  "powerRestrictions": "Code de restrictions & classe de puissance",
+  "powerRestrictionsInfos": "Code de restrictions",
```
Comment (clarani) [inline front/public/locales/fr/rollingstock.json]:
```suggestion
  "powerRestrictions": "Codes de restriction & classes de puissance",
  "powerRestrictionsInfos": "Code de restriction",
```

File: front/public/locales/en/rollingstock.json [hunk 7]
```diff
@@ -78,7 +78,8 @@
   "notLocked": "Not locked",
   "number": "Number",
   "powerClass": "Power class",
-  "powerRestrictions": "Power restrictions",
+  "powerRestrictions": "Power restrictions & power class",
+  "powerRestrictionsInfos": "Power restrictions",
```
Comment (clarani) [inline front/public/locales/en/rollingstock.json]:
```suggestion
  "powerRestrictionsInfos": "Power restriction",
```

**Review by clarani — CHANGES_REQUESTED** [review]
Left some comments

File: front/src/common/SelectorSNCF.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/common/SelectorSNCF.tsx]: Yes totally ! Done.

File: front/src/common/SelectorSNCF.tsx (hunk 2 above)
Comment (SharglutDev) [reply front/src/common/SelectorSNCF.tsx]: Done

File: front/src/common/SelectorSNCF.tsx (hunk 3 above)
Comment (SharglutDev) [reply front/src/common/SelectorSNCF.tsx]: Done

File: front/src/modules/rollingStock/consts.ts (hunk 4 above)
Comment (SharglutDev) [reply front/src/modules/rollingStock/consts.ts:64]: Done

File: front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorCurves.tsx (hunk 5 above)
Comment (SharglutDev) [reply front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorCurves.tsx:438]: Done with adding `as string` cast in the next function

File: front/public/locales/fr/rollingstock.json (hunk 6 above)
Comment (SharglutDev) [reply front/public/locales/fr/rollingstock.json]: Done

File: front/public/locales/en/rollingstock.json (hunk 7 above)
Comment (SharglutDev) [reply front/public/locales/en/rollingstock.json]: Done

File: front/src/common/SelectorSNCF.tsx (hunk 1 above)
Comment (clarani) [reply front/src/common/SelectorSNCF.tsx]: Thanks :pray:

**Review by clarani — APPROVED** [review]
LGTM and tested in local :heavy_check_mark:

**Pull request merged**